integration/test_ml.py

```diff
@@ -0,0 +1,361 @@
+
+    all_models = ml.list_models(list_filter=filter_str)
+    all_model_ids = [mdl.model_id for mdl in all_models.iterate_all()]
+    for mdl in model_list:
+        assert mdl.model_id in all_model_ids
+
+
+def test_list_models_invalid_filter():
+    invalid_filter = 'InvalidFilterParam=123'
+
+    with pytest.raises(exceptions.InvalidArgumentError) as excinfo:
+        ml.list_models(list_filter=invalid_filter)
+    check_firebase_error(excinfo, 400, 'Request contains an invalid argument.')
+
+
+@pytest.mark.parametrize('firebase_model', [NAME_ONLY_ARGS], indirect=True)
+def test_delete_model(firebase_model):
+    ml.delete_model(firebase_model.model_id)
+
+    # Second delete of same model will fail
+    with pytest.raises(exceptions.NotFoundError) as excinfo:
+        ml.delete_model(firebase_model.model_id)
+    check_firebase_error(excinfo, 404, 'Requested entity was not found.')
+
+
+# Test tensor flow conversion functions if tensor flow is enabled.
+#'pip install tensorflow' in the environment if you want _TF_ENABLED = True
+#'pip install tensorflow==2.0.0b' for version 2 etc.
+
+
+def _clean_up_directory(save_dir):
+    if save_dir.startswith(tempfile.gettempdir()) and os.path.exists(save_dir):
+        shutil.rmtree(save_dir)
+
+
+@pytest.fixture
+def keras_model():
+    assert _TF_ENABLED
+    x_array = [-1, 0, 1, 2, 3, 4]
+    y_array = [-3, -1, 1, 3, 5, 7]
+    model = tf.keras.models.Sequential(
+        [tf.keras.layers.Dense(units=1, input_shape=[1])])
+    model.compile(optimizer='sgd', loss='mean_squared_error')
+    model.fit(x_array, y_array, epochs=3)
+    return model
+
+
+@pytest.fixture
+def saved_model_dir(keras_model):
+    assert _TF_ENABLED
+    # different versions have different model conversion capability
+    # pick something that works for each version
+    parent = tempfile.mkdtemp()
+    save_dir = os.path.join(parent, 'child')
+    if tf.version.VERSION.startswith('1.'):
+        tf.reset_default_graph()
+        x_var = tf.placeholder(tf.float32, (None, 3), name="x")
+        y_var = tf.multiply(x_var, x_var, name="y")
+        with tf.Session() as sess:
+            tf.saved_model.simple_save(sess, save_dir, {"x": x_var}, {"y": y_var})
+    else:
+        # If it's not version 1.x or version 2.x we need to update the test.
+        assert tf.version.VERSION.startswith('2.')
+        tf.saved_model.save(keras_model, save_dir)
+    yield save_dir
+    _clean_up_directory(parent)
+
+
+@pytest.mark.skipif(not _TF_ENABLED, reason='Tensor flow is required for this test.')
+def test_from_keras_model(keras_model):
+    source = ml.TFLiteGCSModelSource.from_keras_model(keras_model, 'model2.tflite')
+    assert re.search(
+        '^gs://.*/Firebase/ML/Models/model2.tflite$',
+        source.gcs_tflite_uri) is not None
+
+    # Validate the conversion by creating a model
+    model_format = ml.TFLiteFormat(model_source=source)
+    model = ml.Model(display_name="KerasModel1", model_format=model_format)
```

Comment:
Should probably use a random identifier here too (I'm assuming display_name is a unique key in this API).

Reply:
done